### ISSUE TITLE
Wrong error message in team creation

### ISSUE THREAD

**Issue by dlahlou** [mattermost/mattermost#8845]
On mattermost Api, when I try to create a team with a name containing a capital letter, I get the following error message (in French) : "Le nom doit être composé d'au moins 2 caractères alphanumériques minuscules".
For example, the name "team01" is OK, but "Team01" is KO.
If your API does not allow creating a team with a name containing a capital letter, the error message should be changed to mention that.

Thank you!

**Comment by amyblais** [comment]
Hi @dlahlou!

Thank you for reporting this.

Does this issue repro only on Api? I did a simple test on our [nightly build server](https://pre-release.mattermost.com/signup_user_complete/?id=f1924a8db44ff3bb41c96424cdc20676) and I wasn't able to repro.

Can you also let me know what server version you are using?

**Comment by dlahlou** [comment]
Hi @amyblais 
Yes, the error occurs only on the api.
I'm using Mattermost Version: 4.9.0.

**Comment by amyblais** [comment]
Hi @dlahlou,

Apologies for my bad French skills, but I just want to confirm I understand the expected - in the error message you provided, if I understand correctly, the "caractères alphanumériques minuscules" means "lowercase characters". Can you help with a simple example text of what the error message could say instead?

I can also take a look at editing our API documentation to mention that team names should be in lowercase.

Let me know if I misunderstood anything!

**Comment by dlahlou** [comment]
It depends. From the API, are you willing to allow creating a team with a name containing an uppercase letter? for example "Team01".

**Comment by amyblais** [comment]
@dlahlou, Good question, I will check with the team about that!

**Comment by amyblais** [comment]
Hey @dlahlou, I got feedback from the team:

The Team's "name" field must be lowercase at this time, but the "display_name" can be either case.
The "name" refers to the snippet in the URL, not the one shown in the UI, so we've decided that we want that to be lowercase. Here are docs for reference: https://api.mattermost.com/#tag/teams%2Fpaths%2F~1teams%2Fpost.

Does this describe what you are seeing?

Let me know if this helps or if you still experience any issues!
